src/Adapter/MSTest.TestAdapter/Execution/TestClassInfo.cs

```diff
@@ -303,7 +303,6 @@ public void RunClassInitialize(TestContext testContext)
                     finally
                     {
                         IsClassInitializeExecuted = true;
-                        Debug.Assert(testContext is ITestContext, "Test context doesn't implement ITestContext.");
```

Comment:
Remove assertion because in some tests we are mocking the test context leading to this being false.